File: 645-set-mismatch/set-mismatch.cpp
class Solution {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        vector<int> freq(n+1,0);
        for(int i=0;i<n;i++)
        freq[nums[i]]++;
        int d,m;
        for(int i=1;i<=n;i++){
            if(freq[i]==2) d=i;
            if(freq[i]==0) m = i; 
        }
        return {d,m};
    }
};

File: 645-set-mismatch/set-mismatch.py
class Solution(object):
    def findErrorNums(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        n = len(nums)
        freq = [0]*(n+1)
        
        for i in nums:
            freq[i] +=1
        for i in range(1,n+1):
            if freq[i]==2:
                d=i
            if freq[i] == 0:
                m = i
        return [d,m]